vllm/v1/core/scheduler.py

```diff
@@ -195,8 +196,13 @@ def schedule(self) -> "SchedulerOutput":
                                              request.num_computed_tokens -
                                              request.num_tokens)
                 if num_scheduled_spec_tokens > 0:
+                    if isinstance(request.spec_token_ids, list):
+                        del request.spec_token_ids[num_scheduled_spec_tokens:]
```

Comment:
In practice this branch will always be taken because it will be a list or empty tuple.